sljit_src/sljitLir.c

```diff
@@ -1380,6 +1397,72 @@ static SLJIT_INLINE CHECK_RETURN_TYPE check_sljit_emit_op1(struct sljit_compiler
 	CHECK_RETURN_OK;
 }
 
+static SLJIT_INLINE CHECK_RETURN_TYPE check_sljit_emit_atomic_op1(struct sljit_compiler *compiler, sljit_s32 op, sljit_s32 unset,
+																sljit_s32 data, sljit_sw dataw) {
+	if (SLJIT_UNLIKELY(compiler->skip_checks)) {
+		compiler->skip_checks = 0;
+		CHECK_RETURN_OK;
+	}
+
+#if (defined SLJIT_ARGUMENT_CHECKS && SLJIT_ARGUMENT_CHECKS)
+	CHECK_ARGUMENT(GET_OPCODE(op) >= SLJIT_ATOMIC_LOAD && GET_OPCODE(op) <= SLJIT_ATOMIC_STORE_U32);
+
+	switch ((op) & ~(SLJIT_SET_Z | VARIABLE_FLAG_MASK)) {
+		case SLJIT_ATOMIC_LOAD:
```

Comment:
I don't think so, because on ARM the separate locking load operations are required for a memory section to be locked, and for the locking store to that memory address to succeed; but a regular MOV would not lock the memory.